app/src/main/java/memphis/myapplication/UI/ViewFriendsFragment.java

```diff
@@ -70,16 +80,9 @@ public void onItemClick(View view, int position) {
         alert.setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
             public void onClick(DialogInterface dialog, int which) {
                 // Continue with delete operation
-                Realm realm = Realm.getDefaultInstance();
-                realm.beginTransaction();
-                User user = realm.where(User.class).equalTo("username", friend).findFirst();
-                user.setFriend(false);
-                // Temporary for testing purposes
-                user.setTrust(false);
-                realm.commitTransaction();
+                User user = databaseViewModel.deleteFriendship(friend);
                 Globals.consumerManager.removeConsumer(user.getNamespace());
```

Comment:
I don't really understand the root cause, but the app crashes when you try to remove a friend because it looks for their prefix without a beginning '/', but the friend was added to the consumers list with it. Both functions are called with a string returned from the User.getNamespace() function.
```
07-01 16:01:19.145 8064-8064/? D/Log: 167 : ConsumerManager : createConsumer: Adding friend /edu/Memphis/npChat/mw as consumer
07-01 16:11:22.821 8064-8064/memphis.myapplication D/Log: 174 : ConsumerManager : removeConsumer: Removing edu/Memphis/npChat/mw as consumer

Reply:
It's weird that the namespaces are different! I tried several combinations but couldn't generate the error. However, I found that the UI isn't updated on deleting a friend. I will fix that and update the PR.

Reply:
I think the problem with the mutual friend not showing up is twofold.
First, you're comparing an arraylist of strings with one of users. with "myFriendsList.retainAll(newFriendsList);"
Second, I don't thin the User object actually retains the friendslist from the UserRealm object.
Calling getFriendsofFriend returns nothing.

I can't recreate the problem I had with removing a friend anymore, so don't worry about that for now.

As I mentioned in Slack, the problem where the user doesn't recognize themselves in their friend's friends list is probably an issue with my code, so I'll look into that.

Reply:
I think the problem with the mutual friend not showing up is twofold.
First, you're comparing an arraylist of strings with one of users. with "myFriendsList.retainAll(newFriendsList);"
Second, I don't thin the User object actually retains the friendslist from the UserRealm object.
Calling getFriendsofFriend returns nothing.

I can't recreate the problem I had with removing a friend anymore, so don't worry about that for now.

As I mentioned in Slack, the problem where the user doesn't recognize themselves in their friend's friends list is probably an issue with my code, so I'll look into that.